### PR TITLE
Allow to filter the available tags

### PR DESCRIPTION
**Pull request by nickvergessen**
Required to migrate behaviour like this
https://github.com/nextcloud/files_retention/blob/dcb68844ab563c261d3f8f8be5c90d2e3c9c8407/js/admin.js#L140-L143
to the Vue component.

### CODE REVIEW

**Comment by raimund-schluessler** [comment]
Hm, interesting. We have a related PR here https://github.com/nextcloud/nextcloud-vue/pull/2885 which aims at removing the limit of shown tags of currently 5.

How about we solve #2884 here as well, by removing `:options-limit="5"` and setting the default filter to slice the array to 5 elements to make it non-breaking? @nickvergessen @R0Wi

File: src/components/MultiselectTags/MultiselectTags.vue [hunk 1]
```diff
@@ -65,13 +65,13 @@ export default {
 
 <template>
 	<Multiselect :value="inputValue"
-		:options="tags"
+		:options="availableOptions"
 		:options-limit="5"
```
Comment (raimund-schluessler) [inline src/components/MultiselectTags/MultiselectTags.vue:69]:
```suggestion
```

File: src/components/MultiselectTags/MultiselectTags.vue [hunk 2]
```diff
@@ -113,6 +113,10 @@ export default {
 			type: Boolean,
 			default: true,
 		},
+		filter: {
+			type: Function,
+			default: () => true,
```
Comment (raimund-schluessler) [inline src/components/MultiselectTags/MultiselectTags.vue:118]:
```suggestion
			default: (element, index) => index < 5,
```

**Comment by nickvergessen** [comment]
Would be fine by me

**Comment by R0Wi** [comment]
Nice solution for me, too, thanks! I think the only things missing are the docs and maybe some tests? If you're interested i could provide them in a separate PR

**Review by raimund-schluessler — APPROVED** [review]

**Comment by raimund-schluessler** [comment]
> Nice solution for me, too, thanks! I think the only things missing are the docs and maybe some tests? If you're interested i could provide them in a separate PR

That would be very welcome, I think.

**Comment by nickvergessen** [comment]
@R0Wi very much welcome as I'm very bad in JS/docs and more a consumer only

**Comment by nickvergessen** [comment]
Since this is non breaking I would then also backport it.

So you can also add the docs and tests here if you got the time

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**